Compilation Errors with BW16 (RTL8720DN) Board on Arduino 1.8.19

Hi,

Even after installing the libraries for the BW16 (following the installation guide from Realtek / Ameba IoT (https://www.amebaiot.com/en/amebad-bw16-arduino-getting-started) and selecting the correct board, the project is still failing to compile on the BW16 (RTL8720DN). Below are the warnings and errors encountered during compilation:

Warnings:
WARNING: Category '' in library AmebaFatfsSDIO is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library AmebaFlashMemory is not valid. Setting to 'Uncategorized'
WARNING: Category 'PowerSave' in library AmebaPowerSave is not valid. Setting to 'Uncategorized'
WARNING: Category 'Realtime Clock (RTC)' in library AmebaRTC is not valid. Setting to 'Uncategorized'
WARNING: Category 'Peripheral' in library AmebaWS2812B is not valid. Setting to 'Uncategorized'
WARNING: Category '' in library AmebaWatchdog is not valid. Setting to 'Uncategorized'
Errors:
In file included from /home/atmo/.arduino15/packages/realtek/tools/ameba_d_asdk_toolchain/1.0.1/arm-none-eabi/include/c++/6.5.0/vector:60:0,
                 from /home/atmo/ino/evil-bw16/Evil-BW16-v1-0-2/Evil-BW16-v1-0-2.ino:34:
/home/atmo/.arduino15/packages/realtek/tools/ameba_d_asdk_toolchain/1.0.1/arm-none-eabi/include/c++/6.5.0/bits/stl_algobase.h:243:56: error: macro "min" passed 3 arguments, but takes just 2
     min(const _Tp& __a, const _Tp& __b, _Compare __comp)
                                                        ^
/home/atmo/.arduino15/packages/realtek/tools/ameba_d_asdk_toolchain/1.0.1/arm-none-eabi/include/c++/6.5.0/bits/stl_algobase.h:265:56: error: macro "max" passed 3 arguments, but takes just 2
     max(const _Tp& __a, const _Tp& __b, _Compare __comp)
                                                        ^
Additional Information:

The following libraries were found for WiFi.h:
Used: /home/atmo/.arduino15/packages/realtek/hardware/AmebaD/3.1.3/libraries/WiFi
Not used: /home/atmo/Arduino/libraries/WiFiEspAT

This issue prevents successful compilation for the BW16 (RTL8720DN) board.

You can help ?

--- a/Evil-BW16-v1-0-2.ino
+++ b/Evil-BW16-v1-0-2.ino
@@ -30,12 +30,20 @@
 */
 
 #include <Arduino.h>
-#include <vector>
 #include "wifi_conf.h"
 #include "wifi_util.h"
 #include "wifi_structures.h"
 #include "WiFi.h"
 #include "platform_stdlib.h"
+
+#ifdef max
+#undef max
+#endif
+#ifdef min
+#undef min
+#endif
+
+#include <vector>
 
 //==========================
 // User Configuration

